### DISCUSSION TITLE
synonym for delayWhen as waitFor?

### DISCUSSION THREAD

**Discussion by eliprand** [ReactiveX/rxjs#6019] [Ideas / Feature request]
This is just a suggestion, but `delayWhen()` could use a synonym (or a rename?) as `waitFor()` or `waitUntil()`. For the second time in a couple of years, I had the need to _wait for an observable to complete but ignore its return value and instead use the previous value in the pipe()_. Something like:
```typescript
start().pipe(
  delayWhen(()=>ignoreReturn()),  // <-- ignore the return value, but wait for it to complete
  map((x) => doSomethingWith(x))  // <-- x is the return value from start()
)
```
Both times it took me way too long to figure out that `delayWhen()` wasn't like `delay()` and had very little to do with time (which is what I think about when I hear _delay_). Someone on SO bailed me out today. And it reminded me that a couple of years back, I had created just that, 

The above code would read so much better with `waitFor()`:
```typescript
start().pipe(
  waitFor(()=>ignoreReturn()),
  map((x) => doSomethingWith(x))
)
```

**Comment by Sawtaytoes** [comment]
This is my first time seeing `delayWhen`, but it looks just like `delay` to me. Once a value comes in, it gets delayed until the `notifier$` emits (the return value). Once it emits, that value is passed through the pipeline.

`delay` does the same thing, but its `notifier$` is a fixed `timer` observable with the time you passed into `delay`.

This is the same way `bufferWhen` and others work. There's the prefix or base operator name, then either `Time`, `When`, or `Until` postfixes. It's the same for `audit`, `debounce`, `sample`, and `throttle` too. Strangely though, `delay` is actually `delayTime` based on this naming scheme.

I think naming it `waitFor` would be extremely confusing considering it's just a `delay` operator that takes a `notifier$` and there's nothing in RxJS that uses that nomenclature.

I'm not saying the RxJS naming scheme is great, I'm saying what you're proposing doesn't fit with it. If you really wanted to alias it, which would confusing to other developers, I'd recommend doing `import { delayWhen as waitFor } from 'rxjs/operators'` or create your own `./friendlyRxjsOperators.js` for this purpose.

**Reply by eliprand** [reply]
@Sawtaytoes thanks for the feedback. I agree, `waitFor()` definitely does not fit the naming scheme (I had not done my research about this subject). That's why I had suggested `waitUntil()`.
But even `delayUntil()` feels more natural to me. `delayWhen()` implies a condition. In plain english, _delay when something happens_ implies to delay if a condition is true. Therefore, I would expect a _predicate_ as the parameter.
By contrast _delay until something happens_ implies that the delay is always happening, but is lifted once the event is raised. Does that make sense?
Thanks for the idea of centralizing our aliases in a shared file. This will definitely help bring consistency and discoverability to our aliases.
